src/actions/mod.rs

```diff
@@ -447,16 +451,27 @@ impl ActionHandler {
                 return;
             }
         };
-        let config = self.fmt_config.lock().unwrap();
+
+        let mut config = self.fmt_config.lock().unwrap().get_rustfmt_config().clone();
+        let file_lines = format!(r#"[{{
+            "file": "{}",
+            "range": [{}, {}]
+        }}]"#, path.to_str().unwrap(), range.start.line + 1, range.end.line + 1);
+        config.set().file_lines(file_lines.parse().unwrap());
+
         let mut buf = Vec::<u8>::new();
-        match format_input(input, config.get_rustfmt_config(), Some(&mut buf)) {
+        match format_input(input, &config, Some(&mut buf)) {
             Ok((summary, ..)) => {
                 // format_input returns Ok even if there are any errors, i.e., parsing errors.
                 if summary.has_no_errors() {
                     // Note that we don't need to keep the VFS up to date, the client
                     // echos back the change to us.
-                    let range = ls_util::range_from_vfs_file(&self.vfs, path);
                     let text = String::from_utf8(buf).unwrap();
+
+                    // If Rustfmt returns range of text that changed,
+                    // we will be able to pass only range of changed text to the client.
+                    let range = ls_util::range_from_vfs_file(&self.vfs, path);
```

Comment:
Could move and rename this variable and only make one call to `range_from_vfs_file`